examples/plot_metrics_with_error.py

```diff
@@ -0,0 +1,188 @@
+    centre_adjusted_probability = p + z*z / (2*n)
+    adjusted_standard_deviation = np.sqrt((p*(1 - p) + z*z / (4*n)))/np.sqrt(n)
+    lower_bound = (centre_adjusted_probability - z*adjusted_standard_deviation) / denominator
+    upper_bound = (centre_adjusted_probability + z*adjusted_standard_deviation) / denominator
+    return (round(lower_bound, digits), round(upper_bound, digits))
+
+
+def compute_error_metric(metric_value, sample_size, z_score):
+    """ Standard Error Calculation (Binary Classification)
+
+    Assumes infinitely large population,
+    Should be used when the sampling fraction is small.
+    For sampling fraction > 5%, may want to use finite population correction
+    https://en.wikipedia.org/wiki/Margin_of_error
+
+    Note: 
+        Returns absolute error (%)
+    """
+    return z_score*np.sqrt(metric_value*(1.0-metric_value))/np.sqrt(sample_size)
+
+
+def recall_wilson(y_true, y_pred):
+    assert len(y_true) == len(y_pred)
+    tn, fp, fn, tp = confusion_matrix(y_true, y_pred).ravel()
+    bounds = wilson(tp/(tp+fn), tp + fn, digits_of_precision, z_score)
+    return bounds
+
+def recall_normal_err(y_t, y_p):
+    assert len(y_t) == len(y_p)
+    tn, fp, fn, tp = confusion_matrix(y_t, y_p).ravel()
+    error = compute_error_metric(tp/(tp+fn), tp + fn, z_score=z_score)
+    return (error, error)
+
+def accuracy_wilson(y_true, y_pred):
+    assert len(y_true) == len(y_pred)
+    score = accuracy_score(y_true, y_pred)
+    bounds = wilson(score, len(y_true), digits_of_precision, z_score)
+    return bounds
+
+def accuracy_normal_err(y_true, y_pred):
+    assert len(y_true) == len(y_pred)
+    score = accuracy_score(y_true, y_pred)
+    error = compute_error_metric(score, len(y_true), z_score)
+    return (error, error)
+
+
+# %%
+# Classifier
+# ----------
+# Now we use a Decision Tree to make predictions
+#
+
+classifier = DecisionTreeClassifier(min_samples_leaf=10, max_depth=4)
+classifier.fit(X, y_true)
+y_pred = classifier.predict(X)
+
+# %%
+# MetricFrame
+# -----------
+# Now we create a MetricFrame to generate the recall Wilson bounds and recall normal error
+#
+
+# Analyze metrics using MetricFrame
+metrics_dict = {
+    'Recall': recall_score,
+    'Recall Bounds': recall_wilson,
+    'Recall Error': recall_normal_err,
+    'Accuracy': accuracy_score,
+    'Accuracy Bounds': accuracy_wilson,
+    'Accuracy Error': accuracy_normal_err,
+}
+metric_frame = MetricFrame(metrics_dict, y_true, y_pred, sensitive_features=sex)
+
+# %%
+# plot metrics with (symmetric) error bars
+plot_metric_frame(metric_frame, plot_type="scatter", 
+    metrics=['Recall', 'Accuracy'], 
+    error_bars=['Recall Error', 'Accuracy Error'])
+plot_metric_frame(metric_frame, plot_type="bar", metrics='Recall', error_bars='Recall Error')
+
+# %%
+# plot metrics with confidence intervals (possibly asymmetric)
+plot_metric_frame(metric_frame, plot_type="bar", metrics=['Recall', 'Accuracy'], conf_intervals=['Recall Bounds', 'Accuracy Bounds'])
```

Comment:
The naming of `conf_intervals` combined with `error_bars` is a bit confusing to me. Both of these return a confidence interval plotted as error bars. 

Apart from renaming, we could also combine these into one argument, similar to `matplotlib.pyplot.errorbar`, where `xerr` and `yerr` can take in either scalars or array-likes. In our case we could use allow for a column with scalars for symmetric errors and a column with tuples `(lower bound, upper bound)` which would be interpreted as pre-computed bounds.

Reply:
Combining into one argument, where it could either be symmetric errors if they are scalars and precomputed bounds if they are tuples, doesn't allow for asymmetric errors, right? I only provided an example of symmetric errors but users could also provide asymmetric errors, unless we don't need to support that?

Reply:
From your comment I think I have misinterpreted the meaning of bounds. Don't the bounds indicate the min and max of the confidence interval? Why would that be a problem for asymmetric intervals?

@MiroDudik mentioned asymmetric errors before. I'm assuming that means had a use case in mind for those so we should probably include them.

Reply:
I guess this may be unclear (and maybe we can rename/clarify some places) but when I mention symmetric/asymmetric error, I'm talking about errors relative to the baseline i.e. an asymmetric error of [0.1, 0.2] where the baseline metric is 0.5 would be [0.5 - 0.1, 0.5 + 0.2]. While the bounds would be more directly [0.4, 0.7]. If we support all 3: symmetric error (can be scalar), asymmetric error (tuple), and asymmetric/symmetric bounds (tuple), then we can't do what you mentioned.